cmd/openshift-tuned.go

```diff
@@ -652,36 +664,48 @@ func pullLabels(clientset *kubernetes.Clientset, tuned *tunedState, nodeName str
 		tuned.change.node = true
 	}
 
-	podLabels, err := podLabelsGet(clientset, nodeName)
-	if err != nil {
-		return err
-	}
-	if !reflect.DeepEqual(podLabels, tuned.podLabels) {
-		tuned.podLabels = podLabels
-		tuned.change.pod = true
+	nowUnix := time.Now().Unix()
+	if nowUnix >= tuned.podLabelsPullTime {
+		glog.V(2).Infof("Pulling pod labels")
+		// Pull for pod labels was done >= resyncPeriodPodDefault
+		podLabels, err := podLabelsGet(clientset, nodeName)
+		if err != nil {
+			return err
+		}
+		tuned.podLabelsPullTime = nowUnix + resyncPeriodPodDefault + getJitter(resyncPeriodPodDefault, 0.3) // try to avoid lockstep
+		if !reflect.DeepEqual(podLabels, tuned.podLabels) {
+			tuned.podLabels = podLabels
+			tuned.change.pod = true
+		}
 	}
+
 	return nil
 }
 
 func pullResyncPeriod() int64 {
 	var (
 		err                  error
-		resyncPeriodDuration int64 = resyncPeriodDefault
+		resyncPeriodDuration int64 = resyncPeriodNodeDefault
 	)
 	if os.Getenv("RESYNC_PERIOD") != "" {
 		resyncPeriodDuration, err = strconv.ParseInt(os.Getenv("RESYNC_PERIOD"), 10, 64)
 		if err != nil {
-			glog.Errorf("Error: cannot parse RESYNC_PERIOD (%s), using %d", os.Getenv("RESYNC_PERIOD"), resyncPeriodDefault)
-			resyncPeriodDuration = resyncPeriodDefault
+			glog.Errorf("Error: cannot parse RESYNC_PERIOD (%s), using %d", os.Getenv("RESYNC_PERIOD"), resyncPeriodNodeDefault)
+			resyncPeriodDuration = resyncPeriodNodeDefault
 		}
 	}
 
-	// Add some randomness to the resync period, so that we don't end up in a lockstep querying the API server
-	resyncPeriodDuration += rand.Int63n(resyncPeriodDuration/5+1) - resyncPeriodDuration/10
 	return resyncPeriodDuration
 }
 
-func changeWatcher() (err error) {
+func pullResyncPeriodWithJitter() int64 {
+	resyncPeriodDuration := pullResyncPeriod()
+
+	// Add some randomness to the resync period, so that we don't end up in lockstep querying the API server
+	return resyncPeriodDuration + getJitter(resyncPeriodDuration, 0.3)
```

Comment:
You've created a variable for one function return value but not the other, maybe do both or neither?

Reply:
I'm a bit confused here